backups/cognito/backup_users.py

```diff
@@ -0,0 +1,63 @@
+import boto3
+import csv
+from datetime import datetime
+from botocore.exceptions import ClientError
+import os
+
+client = boto3.client('cognito-idp')
+s3_client = boto3.client('s3')
+
+def backup_users():
+    try:
+        all_users = get_all_users()
+    except Exception as e:
+        print('Getting current cognito users has failed ', e)
+
+    csv_headers = ['cognito:username','name','given_name',
+    'family_name','middle_name','nickname','preferred_username',
+    'profile','picture','website','email','email_verified',
+    'gender','birthdate','zoneinfo','locale','phone_number',
+    'phone_number_verified','address','updated_at','cognito:mfa_enabled']
+    
+    with open('data.csv','w', encoding='UTF8') as f:
+        writer = csv.writer(f)
+        writer.writerow(csv_headers)
+
+        for user in all_users:
+            name = list(filter(lambda attribute: attribute['Name']=='name', user['Attributes']))[0]['Value']
+            email = list(filter(lambda attribute: attribute['Name']=='email', user['Attributes']))[0]['Value']
+            email_verified = list(filter(lambda attribute: attribute['Name']=='email_verified', user['Attributes']))[0]['Value']
+            
+            new_row = [email,name,'','','','','','','','',email,email_verified,'','','','','','FALSE','','','FALSE']
+            writer.writerow(new_row)
+            file_name = '{}/biomage-userpool-case-insensitive-production/data.csv'.format(datetime.now().isoformat())
+    try:
+        s3_client.upload_file('data.csv', 'biomage-backups-production', file_name)
+        #removing the local file after saving to S3
+        os.remove('data.csv')
+    except ClientError as e:
+        print('Uploading backup to S3 has failed ', e)
+    
```

Comment:
where are the tests that the ticket mentions about to check if the backup is diffferent or the same as the real data?